doc/vlan/SONiC_VLAN_trunk_all_except_none_HLD.md

```diff
@@ -0,0 +1,701 @@
+
+**Scenario 1:**
+<br> Vlan 5 exists. 
+<br> Step 1: sonic(conf-if-EthernetX)#switchport access Vlan 5
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 5
+<br> Result: Port remains untagged in Vlan5, port is tagged in vlan 1-4,6-4094 
+<br> Step 3: sonic(conf-if-EthernetX)#no switchport access vlan 
+<br> Result: Port tagged in vlan 1-4,6-4094 
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-4,6-4094
+sonic#
+```
+
+**Scenario 2:**
+<br> Vlan 10 exists.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport access vlan 10 
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Result: Port remains untagged in vlan 10.
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 10
+ switchport trunk allowed Vlan 1-4094
+sonic#
+```
+
+**Scenario 3:**
+<br> Vlan 10 exists.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 2: sonic(conf-if-EthernetX)#switchport access vlan 10
+<br> Result: Port untagged port in vlan 10.
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 10
+ switchport trunk allowed Vlan 1-4094
+sonic#
+```
+
+**Scenario 4:**
+<br> Vlan 20 does not exist.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport access vlan 20
+<br> Result: Port untagged in vlan 20
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 20
+sonic#
+```
+Verify: Check 'show vlan' output, should be empty since no VLAN exists.  
+```
+sonic# show Vlan
+
+```
+Step 2: Create Vlan20 - sonic(config)#interface vlan 20
+<br> Verify member added during Vlan creation using 'show vlan' output.  
+```
+sonic(conf-if-Vlan20)# do show Vlan
+Q: A - Access (Untagged), T - Tagged
+NUM        Status      Q Ports
+20         Inactive    A  EthernetX
+
+```
+
+**Scenario 5:**
+<br> Vlan 20 does not exist.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 2: sonic(conf-if-EthernetX)#switchport access vlan 20
+<br> Step 3: sonic(config)#interface vlan 20
+<br> Result: Port will be untagged in vlan 20
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 20
+ switchport trunk allowed Vlan 1-4094
```

Comment:
since port will be untagged in vlan20,  the show running config should have following lines
switchport trunk allowed Vlan 1-19,21-4094

Reply:
No, show running config will display same config what is specified in CLI(since "all" was specified 1-4094 is displayed). 
After "no switchport trunk allowed vlan 20" cmd, the show running will display -  1-19,21-4094